### PR TITLE
micro: port operator FILL from lite

### PR DESCRIPTION
**Pull request by rkuester**
For micro, implement the FILL kernel and test, and add both to the build.

This PR contains several commits for readable history and ease of review.

This is PR5 in the sequence of PRs, outlined in issue #45306, to port operator FILL from lite to micro.

### CODE REVIEW

**Comment by rkuester** [comment]
This PR is stacked atop PR, #45646. It should be rebased and moved from draft to ready status as soon as that PR is merged.

The appropriate sequence of PRs and mapping of commits to PRs in this port of operator FILL was discussed in #45612.

File: tensorflow/lite/micro/kernels/fill.cc
Comment (petewarden) [inline tensorflow/lite/micro/kernels/fill.cc:113]: This will need to be replaced with the macro equivalent.

**Review by petewarden — APPROVED** [review]

**Comment by gbaned** [comment]
@rkuester This PR is in draft, can you please take a look on it. Thanks!

**Comment by gbaned** [comment]
@rkuester  This PR is in draft, can you please take a look on it. Thanks!

**Comment by rkuester** [comment]
FWIW, this PR is stacked atop PR #45646 and is awaiting approval thereof before I rebase it and take it out of draft.

**Comment by gbaned** [comment]
@rkuester Can you please resolve conflicts? Thanks!

**Review by petewarden — APPROVED** [review]

**Pull request merged**
